<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>720</width>
    <height>934</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Ignored" vsizetype="Ignored">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_7">
    <property name="leftMargin">
     <number>2</number>
    </property>
    <property name="topMargin">
     <number>2</number>
    </property>
    <property name="rightMargin">
     <number>2</number>
    </property>
    <property name="bottomMargin">
     <number>2</number>
    </property>
    <item row="0" column="0">
     <widget class="QScrollArea" name="scrollArea">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Ignored" vsizetype="Minimum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="mouseTracking">
       <bool>true</bool>
      </property>
      <property name="tabletTracking">
       <bool>true</bool>
      </property>
      <property name="layoutDirection">
       <enum>Qt::LeftToRight</enum>
      </property>
      <property name="styleSheet">
       <string notr="true">QScrollArea {
	border: none;
}

QScrollBar {

	border-radius: 2px;
}

QScrollBar:horizontal {
	height: 20px;
}

QScrollBar:vertical {
	width: 20px;
	position: left;
}

QScrollBar::handle {
	background: #b3b3b3;
	border-radius: 2px;
 }

QScrollBar::handle:horizontal {
	height: 25px;
	min-width: 30px;
}

QScrollBar::handle:vertical {
	width: 25px;
	min-height: 30px;
}

QScrollBar::add-line {
	border: none;
	background: none;
 }

QScrollBar::sub-line {
	border: none;
	background: none;
}</string>
      </property>
      <property name="lineWidth">
       <number>5</number>
      </property>
      <property name="verticalScrollBarPolicy">
       <enum>Qt::ScrollBarAsNeeded</enum>
      </property>
      <property name="horizontalScrollBarPolicy">
       <enum>Qt::ScrollBarAsNeeded</enum>
      </property>
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="scrollAreaWidgetContents">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>696</width>
         <height>1089</height>
        </rect>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_10">
        <item>
         <widget class="QGroupBox" name="groupBox_2">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title">
           <string>BLE Devices</string>
          </property>
          <layout class="QGridLayout" name="gridLayout">
           <item row="2" column="0">
            <spacer name="verticalSpacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="2" column="1">
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="0" column="0">
            <layout class="QGridLayout" name="gridLayout_3">
             <item row="0" column="2">
              <widget class="QCheckBox" name="scanPeriodicallyCheckBox">
               <property name="text">
                <string>Scan Periodically</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLabel" name="scanningIndicatorLabel">
               <property name="minimumSize">
                <size>
                 <width>100</width>
                 <height>0</height>
                </size>
               </property>
               <property name="text">
                <string/>
               </property>
              </widget>
             </item>
             <item row="0" column="3">
              <spacer name="horizontalSpacer_3">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item row="0" column="0">
              <widget class="QLabel" name="discoveryStatusLabel">
               <property name="text">
                <string>Device Discovery Status:</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="1" column="0">
            <widget class="QTableWidget" name="devicesTableWidget">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Ignored" vsizetype="Ignored">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>120</height>
              </size>
             </property>
             <property name="columnCount">
              <number>3</number>
             </property>
             <column/>
             <column/>
             <column/>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="device_groupBox">
          <property name="title">
           <string>Device Control</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <widget class="QPushButton" name="bleConnectPushButton">
             <property name="minimumSize">
              <size>
               <width>97</width>
               <height>0</height>
              </size>
             </property>
             <property name="palette">
              <palette>
               <active>
                <colorrole role="Button">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>108</red>
                   <green>200</green>
                   <blue>80</blue>
                  </color>
                 </brush>
                </colorrole>
               </active>
               <inactive>
                <colorrole role="Button">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>108</red>
                   <green>200</green>
                   <blue>80</blue>
                  </color>
                 </brush>
                </colorrole>
               </inactive>
               <disabled>
                <colorrole role="Button">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>108</red>
                   <green>200</green>
                   <blue>80</blue>
                  </color>
                 </brush>
                </colorrole>
               </disabled>
              </palette>
             </property>
             <property name="text">
              <string>Connect</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="bleDisconnectPushButton">
             <property name="minimumSize">
              <size>
               <width>97</width>
               <height>0</height>
              </size>
             </property>
             <property name="palette">
              <palette>
               <active>
                <colorrole role="Button">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>196</red>
                   <green>86</green>
                   <blue>86</blue>
                  </color>
                 </brush>
                </colorrole>
               </active>
               <inactive>
                <colorrole role="Button">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>196</red>
                   <green>86</green>
                   <blue>86</blue>
                  </color>
                 </brush>
                </colorrole>
               </inactive>
               <disabled>
                <colorrole role="Button">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>196</red>
                   <green>86</green>
                   <blue>86</blue>
                  </color>
                 </brush>
                </colorrole>
               </disabled>
              </palette>
             </property>
             <property name="text">
              <string>Disconnect</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="deviceControlGroupBox">
          <property name="title">
           <string>Services and Characteristics</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <widget class="QTreeWidget" name="bleServicesTreeWidget">
             <property name="sizePolicy">
              <sizepolicy hsizetype="MinimumExpanding" vsizetype="Ignored">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>100</height>
              </size>
             </property>
             <property name="horizontalScrollBarPolicy">
              <enum>Qt::ScrollBarAsNeeded</enum>
             </property>
             <attribute name="headerVisible">
              <bool>false</bool>
             </attribute>
             <column>
              <property name="text">
               <string notr="true">1</string>
              </property>
             </column>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="groupBox">
             <property name="title">
              <string>Characteristics' Control</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_5">
              <item>
               <widget class="QGroupBox" name="write_groupBox">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="minimumSize">
                 <size>
                  <width>0</width>
                  <height>0</height>
                 </size>
                </property>
                <property name="title">
                 <string>WRITE</string>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout_7">
                 <item>
                  <layout class="QVBoxLayout" name="verticalLayout">
                   <item>
                    <layout class="QHBoxLayout" name="horizontalLayout_5">
                     <item>
                      <widget class="QLabel" name="label">
                       <property name="text">
                        <string>HEX</string>
                       </property>
                      </widget>
                     </item>
                     <item>
                      <widget class="QLineEdit" name="bleCharacteristicWriteLineEdit"/>
                     </item>
                     <item>
                      <widget class="QPushButton" name="bleCharacteristicWritePushButton">
                       <property name="minimumSize">
                        <size>
                         <width>97</width>
                         <height>0</height>
                        </size>
                       </property>
                       <property name="text">
                        <string>Write</string>
                       </property>
                      </widget>
                     </item>
                    </layout>
                   </item>
                  </layout>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <widget class="QGroupBox" name="read_groupBox">
                <property name="title">
                 <string>READ / NOTIFY</string>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout_3">
                 <item>
                  <widget class="QGroupBox" name="groupBox_4">
                   <property name="title">
                    <string>Output</string>
                   </property>
                   <layout class="QVBoxLayout" name="verticalLayout_6">
                    <item>
                     <widget class="QPlainTextEdit" name="outputPlainTextEdit">
                      <property name="sizePolicy">
                       <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
                        <horstretch>0</horstretch>
                        <verstretch>0</verstretch>
                       </sizepolicy>
                      </property>
                      <property name="minimumSize">
                       <size>
                        <width>0</width>
                        <height>0</height>
                       </size>
                      </property>
                      <property name="readOnly">
                       <bool>true</bool>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <layout class="QHBoxLayout" name="horizontalLayout_7">
                      <item>
                       <spacer name="horizontalSpacer_4">
                        <property name="orientation">
                         <enum>Qt::Horizontal</enum>
                        </property>
                        <property name="sizeHint" stdset="0">
                         <size>
                          <width>40</width>
                          <height>20</height>
                         </size>
                        </property>
                       </spacer>
                      </item>
                      <item>
                       <widget class="QLabel" name="label_2">
                        <property name="text">
                         <string>Format:</string>
                        </property>
                       </widget>
                      </item>
                      <item>
                       <widget class="QComboBox" name="bleCharacteristicReadTypeComboBox">
                        <property name="minimumSize">
                         <size>
                          <width>97</width>
                          <height>0</height>
                         </size>
                        </property>
                        <item>
                         <property name="text">
                          <string>HEX</string>
                         </property>
                        </item>
                        <item>
                         <property name="text">
                          <string>BIN</string>
                         </property>
                        </item>
                        <item>
                         <property name="text">
                          <string>UNICODE</string>
                         </property>
                        </item>
                       </widget>
                      </item>
                      <item>
                       <widget class="QPushButton" name="clearOutputPushButton">
                        <property name="minimumSize">
                         <size>
                          <width>97</width>
                          <height>0</height>
                         </size>
                        </property>
                        <property name="text">
                         <string>Clear</string>
                        </property>
                       </widget>
                      </item>
                     </layout>
                    </item>
                   </layout>
                  </widget>
                 </item>
                 <item>
                  <widget class="QGroupBox" name="groupBox_5">
                   <property name="title">
                    <string>Control</string>
                   </property>
                   <layout class="QHBoxLayout" name="horizontalLayout_12">
                    <item>
                     <spacer name="horizontalSpacer_6">
                      <property name="orientation">
                       <enum>Qt::Horizontal</enum>
                      </property>
                      <property name="sizeHint" stdset="0">
                       <size>
                        <width>40</width>
                        <height>20</height>
                       </size>
                      </property>
                     </spacer>
                    </item>
                    <item>
                     <widget class="QPushButton" name="listenNotifyPushButton">
                      <property name="text">
                       <string>Listen Notify</string>
                      </property>
                     </widget>
                    </item>
                    <item>
                     <widget class="QPushButton" name="bleCharacteristicReadPushButton">
                      <property name="minimumSize">
                       <size>
                        <width>97</width>
                        <height>0</height>
                       </size>
                      </property>
                      <property name="text">
                       <string>Read</string>
                      </property>
                     </widget>
                    </item>
                   </layout>
                  </widget>
                 </item>
                 <item>
                  <widget class="QGroupBox" name="groupBox_3">
                   <property name="title">
                    <string>Save</string>
                   </property>
                   <layout class="QVBoxLayout" name="verticalLayout_9">
                    <item>
                     <layout class="QHBoxLayout" name="horizontalLayout_9">
                      <item>
                       <widget class="QCheckBox" name="notifyCheckBox">
                        <property name="text">
                         <string>Notify</string>
                        </property>
                       </widget>
                      </item>
                      <item>
                       <widget class="QCheckBox" name="readCheckBox">
                        <property name="text">
                         <string>Read</string>
                        </property>
                       </widget>
                      </item>
                      <item>
                       <spacer name="horizontalSpacer_5">
                        <property name="orientation">
                         <enum>Qt::Horizontal</enum>
                        </property>
                        <property name="sizeHint" stdset="0">
                         <size>
                          <width>40</width>
                          <height>20</height>
                         </size>
                        </property>
                       </spacer>
                      </item>
                     </layout>
                    </item>
                    <item>
                     <layout class="QHBoxLayout" name="horizontalLayout_10">
                      <item>
                       <widget class="QLineEdit" name="saveLineEdit"/>
                      </item>
                      <item>
                       <widget class="QPushButton" name="browsePushButton">
                        <property name="minimumSize">
                         <size>
                          <width>97</width>
                          <height>0</height>
                         </size>
                        </property>
                        <property name="text">
                         <string>Browse</string>
                        </property>
                       </widget>
                      </item>
                     </layout>
                    </item>
                   </layout>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>720</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
